src/core_plugins/metrics/public/kbn_vis_types/request_handler.js

```diff
@@ -8,25 +8,25 @@ const MetricsRequestHandlerProvider = function (Private, Notifier, config, timef
 
   return {
     name: 'metrics',
-    handler: function (vis, appState, uiState) {
+    handler: function (vis, { uiState, timeRange }) {
```

Comment:
Make TSVB request handler only use the passed timeRange and not looking into timefilter directly anymore.